scRNA-seq-advanced/02-celltype_assignment.Rmd

```diff
@@ -44,149 +49,239 @@ set.seed(12345)
 adt_df <- logcounts(altExp(sce)) %>%
-  t() %>% 
-  as.matrix() %>%
-  as.data.frame()
+  t() %>% # transpose
+  as.matrix() %>% # convert to matrix
+  as.data.frame() # convert to data frame
+
+# view the data frame
+head(adt_df)
 ```
 
-With a bit more wrangling, we can put the expression data for the three tags we care about into a single column, which will allow us to plot all three together a bit more neatly.
-Here we are using the `tidyr::pivot_longer()` function, which allows us to convert our data frame with one column per tag into a data frame with separate columns for the tag id (`ADT`) and the expression value (`logcount`).
+If we just wanted to plot one of these tags, we could do so right away, but with a bit more data wrangling, we can convert these results into a "tidier" format, that will allow us to take full advantage of `tidyverse` tools!
+In particular, it will let us plot them all at once with `ggplot2` faceting.
 
-```{r}
+Right now the data is in a "wide" format, such that each column is a different tag. 
+But the data in all of the columns is the same type, and measures something similar: the normalized count of an ADT.
+One could even argue that each row contains 10 different observations, where the "tidy" data ideal, as espoused by [Wickham (2014)](https://doi.org/10.18637/jss.v059.i10), requires a single observation per row, a "long" format.
+This long format will have one column that tells us which ADT was measured and a second column with the measurement value itself.
+
+We can perform this conversion using the [`tidyr::pivot_longer()`](https://tidyr.tidyverse.org/articles/pivot.html) function,
+which allows us to convert our data frame with one column per tag into a data frame with separate columns for the tag id (`ADT`) and the expression value (`logcount`).
+Following conversion, we will filter to just the ADTs that we care about.
+
+```{r pivot longer, live=TRUE}
 adt_df_long <- adt_df %>%
-  dplyr::select(c("CD3", "CD4", "CD8")) %>%
-  tidyr::pivot_longer(everything(), names_to = "ADT", values_to = "logcount")
-```
+  # pivot to long format
+  tidyr::pivot_longer(
+    everything(), # use all columns
+    names_to = "ADT", # convert row names to a column called "ADT"
+    values_to = "logcount" # name the value column "logcount"
+  ) %>%
+  # filter to tags we are interested in
+  dplyr::filter(ADT %in% c("CD3", "CD4", "CD8"))
 
+# look at the resulting df
+head(adt_df_long)
+```
 
-Now we can make a density plot with `ggplot2`!
+Now we can make a density plot with `ggplot2` for all three ADTs we are interested in at once.
 
-```{r}
+```{r plot ADTs, live=TRUE}
+# plot logcounts by ADT
 ggplot(adt_df_long, aes(x = logcount, fill = ADT)) + 
-  geom_density() +
-  scale_x_continuous(breaks = seq(0, 10, by = 2)) + 
-  facet_wrap(~ADT, ncol = 1) +
-  theme_bw() +
-  theme(legend.position = "none")
+  geom_density() + # density plot
+  facet_wrap(~ADT, ncol = 1) + # facet by ADT
```

Comment:
```suggestion
  facet_wrap(vars(ADT), ncol = 1) + # facet by ADT
```

Reply:
we're headed this direction I think?

Reply:
We are, but I think I'm going to go all the way to `facet_grid` here, as the explicit rows is nice:
```suggestion
  facet_grid(rows = vars(ADT)) + # facet by ADT
```